### DISCUSSION TITLE
Question: image node usage

### DISCUSSION THREAD

**Discussion by subhrajit-akrabhi** [facebook/lexical#2835] [Q&A] [answered]
I am working on blog like web project using lexical editor in nextjs. I am trying to implement image plugin in lexical editor. For now it coverts to base64 format which will be expensive for storing in database. Instead of storing base64 I want to store the image link along with other text inside database. Also I am facing issue to implement image plugin of lexical editor in nextjs.

steps I want to follow

I want to upload image in aws s3 and then want to show in editor as image.
Another way I am trying to implement like a modal will open where image can be selected and it uploads to aws s3 and gets image url, then want to show image url in editor.
Is there any to implement?

**Comment by fantactuka** [comment, marked as answer]
See playground image node as an example https://github.com/facebook/lexical/blob/3aa15b7b6c48f9aac3297445240d1849c26b462b/packages/lexical-playground/src/nodes/ImageNode.tsx#L121

As you mentioned, use a modal to upload and get image url and then insert image node using that url. It's very similar to what playground does

**Reply by tomprats** [reply]
@subhrajit-akrabhi Did you manage to do this? I'm looking into editors and hoping to find one that is simple to upload things with
